examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,262 @@
+
+clf_list = [(lr, 'Logistic'),
+            (gnb, 'Naive Bayes'),
+            (gnb_isotonic, 'Naive Bayes + Isotonic'),
+            (gnb_sigmoid, 'Naive Bayes + Sigmoid')]
+
+fig = plt.figure(figsize=(10, 10))
+gs = GridSpec(4, 2)
+colors = plt.cm.get_cmap('Dark2')
 
-def plot_calibration_curve(est, name, fig_index):
-    """Plot calibration curve for est w/o and with calibration. """
-    # Calibrated with isotonic calibration
-    isotonic = CalibratedClassifierCV(est, cv=2, method='isotonic')
+ax1 = fig.add_subplot(gs[:2, :2])
+viz_objects = {}
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    viz = plot_calibration_curve(
+        clf, X_test, y_test, n_bins=10, name=name, ax=ax1, color=colors(i)
+    )
+    viz_objects[name] = viz
+
+ax1.grid()
+ax1.set_title('Calibration plots (Naive Bayes)')
+ax1.set(xlabel="")
+
+# Add histogram
+grid_positions = [(2, 0), (2, 1), (3, 0), (3, 1)]
+for i, (_, name) in enumerate(clf_list):
+    row, col = grid_positions[i]
+    ax = fig.add_subplot(gs[row, col])
+
+    ax.hist(
+        viz_objects[name].y_prob, range=(0, 1), bins=10, label=name,
+        color=colors(i)
+    )
+    ax.set(title=name, xlabel="Mean predicted probability", ylabel="Count")
+
+plt.tight_layout()
+plt.show()
 
-    # Calibrated with sigmoid calibration
-    sigmoid = CalibratedClassifierCV(est, cv=2, method='sigmoid')
+# %%
+# Raw :class:`~sklearn.naive_bayes.GaussianNB` performs very badly because of
+# the redundant features which violate the assumption of feature-independence
+# and result in an overly confident classifier, which is indicated by the
+# typical transposed-sigmoid curve. Calibration of the probabilities of
+# :class:`~sklearn.naive_bayes.GaussianNB` with isotonic regression can fix
+# this issue as can be seen from the nearly diagonal calibration curve. Sigmoid
+# regression also improves calibration slightly,
+# albeit not as strongly as the non-parametric isotonic regression. This can be
+# attributed to the fact that we have plenty of calibration data such that the
+# greater flexibility of the non-parametric model can be exploited.
+#
+# Below we show the Brier score , precision, recall and F1 score (see
+# :ref:`User Guide <precision_recall_f_measure_metrics>`). Notice that
+# although calibration improves the Brier score (a metric composed of
+# calibration loss and refinement loss), it does not significantly
+# alter the prediction accuracy measures (precision, recall and F1 score).
+# This is because calibration should not significantly move at the location of
+# the decision threshold (at x = 0.5 on the graph). Calibration should
+# however, make the predicted probabilities more accurate and thus more useful
+# for making allocation decisions under uncertainty.
 
-    # Logistic regression with no calibration as baseline
-    lr = LogisticRegression(C=1.)
+index = {}
+brier = []
+precision = []
+recall = []
+f1 = []
```

Comment:
Could you try to add the log-loss to this example? I believe this is another reasonable calibration dependent classification metric for binary classification.